File: packages/mediation/gma_mediation_dtexchange/android/src/main/kotlin/io/flutter/plugins/googlemobileads/mediation/gma_mediation_dtexchange/GmaMediationDTExchangePlugin.kt
package io.flutter.plugins.googlemobileads.mediation.gma_mediation_dtexchange

import com.fyber.inneractive.sdk.external.InneractiveAdManager
import io.flutter.embedding.engine.plugins.FlutterPlugin

/** Manages DTExchangePrivacyApi and implements the needed methods. */
class GmaMediationDTExchangePlugin: FlutterPlugin, DTExchangePrivacyApi {
  override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
    DTExchangePrivacyApi.setUp(flutterPluginBinding.binaryMessenger, this)
  }

  override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
    DTExchangePrivacyApi.setUp(binding.binaryMessenger, null)
  }

  override fun setLgpdConsent(wasConsentGiven: Boolean) {
    InneractiveAdManager.setLgpdConsent(wasConsentGiven)
  }

  override fun clearLgpdConsentData() {
    InneractiveAdManager.clearLgpdConsentData()
  }

  override fun setUSPrivacyString(usPrivacyString: String) {
    InneractiveAdManager.setUSPrivacyString(usPrivacyString)
  }

  override fun clearUSPrivacyString() {
    InneractiveAdManager.clearUSPrivacyString()
  }
}

File: packages/mediation/gma_mediation_dtexchange/ios/Classes/GmaMediationDTExchangePlugin.swift
import Flutter
import IASDKCore
import UIKit

/// Manages DTExchangePrivacyApi and implements the needed methods.
public class GmaMediationDTExchangePlugin: NSObject, FlutterPlugin, DTExchangePrivacyApi {
  let dtExchangeApi: DTExchangePrivacyProtocol

  init (dtExchangeApi: DTExchangePrivacyProtocol) {
    self.dtExchangeApi = dtExchangeApi
  }

  public static func register(with registrar: FlutterPluginRegistrar) {
    let messenger : FlutterBinaryMessenger = registrar.messenger()
    let api : DTExchangePrivacyApi&NSObjectProtocol=GmaMediationDTExchangePlugin.init(dtExchangeApi: DTExchangePrivacyImpl())
    DTExchangePrivacyApiSetup.setUp(binaryMessenger: messenger, api: api)
  }
  public func detachFromEngine(for registrar: FlutterPluginRegistrar) {
    let messenger : FlutterBinaryMessenger = registrar.messenger()
    DTExchangePrivacyApiSetup.setUp(binaryMessenger: messenger, api: nil)
  }

  func setLgpdConsent(wasConsentGiven: Bool) throws {
    dtExchangeApi.setLgpdConsent(wasConsentGiven: wasConsentGiven)
  }

  func clearLgpdConsentData() throws {
    dtExchangeApi.clearLgpdConsentData()
  }

  func setUSPrivacyString(usPrivacyString: String) throws {
    dtExchangeApi.setUSPrivacyString(usPrivacyString: usPrivacyString)
  }

  func clearUSPrivacyString() throws {
    dtExchangeApi.clearUSPrivacyString()
  }
}

protocol DTExchangePrivacyProtocol {
  func setLgpdConsent(wasConsentGiven: Bool)

  func clearLgpdConsentData()

  func setUSPrivacyString(usPrivacyString: String)

  func clearUSPrivacyString()
}

class DTExchangePrivacyImpl : DTExchangePrivacyProtocol {
  func setLgpdConsent(wasConsentGiven: Bool) {
    IASDKCore.sharedInstance().lgpdConsent = wasConsentGiven ? IALGPDConsentType.given : IALGPDConsentType.denied
  }
  
  func clearLgpdConsentData() {
    IASDKCore.sharedInstance().clearLGPDConsentData()
  }

  func setUSPrivacyString(usPrivacyString: String) {
    IASDKCore.sharedInstance().ccpaString = usPrivacyString
  }

  func clearUSPrivacyString() {
    IASDKCore.sharedInstance().ccpaString = nil
  }
}
